astacus/coordinator/plugins/base.py

```diff
@@ -152,10 +153,12 @@ class UploadBlocksStep(Step[List[ipc.SnapshotUploadResult]]):
 
     storage_name: str
     validate_file_hashes: bool = True
+    upload_request: str = "upload"
+    list_hexdigests_step: type = ListHexdigestsStep
```

Comment:
The type is too vague here (and I'm surprised mypy doesn't complain), it should be `type[Step[Set[str]]]`, this allows static validatoin that the return value of `context.get_result` has the correct type.




Ah, I know why mypy didn't complain :( 
```python
def build_node_index_datas(
    *, hexdigests, snapshots: List[ipc.SnapshotResult], node_indices: List[int]
) -> List[NodeIndexData]:
```